conn.go

```diff
@@ -384,18 +384,19 @@ func (c *Conn) identify() (*IdentifyResponse, error) {
 }
 
 func (c *Conn) upgradeTLS(tlsConf *tls.Config) error {
-	// create a local copy of the config to set ServerName for this connection
-	var conf tls.Config
-	if tlsConf != nil {
-		conf = *tlsConf
-	}
 	host, _, err := net.SplitHostPort(c.addr)
 	if err != nil {
 		return err
 	}
+
+	// create a local copy of the config to set ServerName for this connection
+	conf := new(tls.Config)
```

Comment:
minor nit: the convention we've followed for cases like this is `conf := &tls.Config{}`

Reply:
Personally I prefer `new` unless I'm also instantiating the types fields at the same time. 

Went ahead and changed it.